pkg/controller/migmigration/backup.go

```diff
@@ -19,47 +18,25 @@ import (
 // Ensure the initial backup on the source cluster has been created
 // and has the proper settings.
 func (t *Task) ensureInitialBackup() (*velero.Backup, error) {
-	newBackup, err := t.buildBackup(nil)
+	client, err := t.getSourceClient()
+	if err != nil {
+		log.Trace(err)
+		return nil, err
+	}
+	newBackup, err := t.buildBackup(client)
 	if err != nil {
 		log.Trace(err)
 		return nil, err
 	}
 	newBackup.Labels[InitialBackupLabel] = t.UID()
 	newBackup.Spec.ExcludedResources = excludedInitialResources
 	delete(newBackup.Annotations, QuiesceAnnotation)
-	foundBackup, err := t.getInitialBackup()
+	err = client.Create(context.TODO(), newBackup)
 	if err != nil {
 		log.Trace(err)
 		return nil, err
 	}
-	if foundBackup == nil {
```

Comment:
I think we still need this block. It makes sure we don't attempt to create the CR if it already exists. We do need to remove the `!t.equalsBackup...` block, though.

Reply:
This block is now executed before opening client and building backup: https://github.com/konveyor/mig-controller/pull/520/files#diff-bae2a9cc4cbb76d893b45a873d56cc30R21-R28 if the backup or restore was already created, then the step will be ended

Reply:
Ahh. I see that now.